subsystem/image_test.go

```diff
@@ -117,73 +115,32 @@ var _ = Describe("system-test proxy update tests", func() {
-		// Generate ISO of registered cluster without proxy configured
-		_, err := userBMClient.Installer.GenerateClusterISO(ctx, &installer.GenerateClusterISOParams{
-			ClusterID:         clusterID,
-			ImageCreateParams: &models.ImageCreateParams{},
-		})
-		Expect(err).NotTo(HaveOccurred())
+		// Generate infraEnv and ISO without proxy configured
+		infraEnvID := registerInfraEnv(&clusterID).ID
 
 		// Update cluster with proxy settings
 		httpProxy := "http://proxyserver:3128"
 		noProxy := "test.com"
-		_, err = userBMClient.Installer.UpdateCluster(ctx, &installer.UpdateClusterParams{
-			ClusterUpdateParams: &models.ClusterUpdateParams{
-				HTTPProxy: &httpProxy,
-				NoProxy:   &noProxy,
-			},
-			ClusterID: clusterID,
-		})
-		Expect(err).ShouldNot(HaveOccurred())
-
-		// Verify proxy settings changed event emitted
-		verifyEventExistence(clusterID, "Proxy settings changed")
-
-		// at least 10s must elapse between requests to generate the same ISO
-		time.Sleep(time.Second * 10)
 
-		// Generate ISO of registered cluster with proxy configured
-		_, err = userBMClient.Installer.GenerateClusterISO(ctx, &installer.GenerateClusterISOParams{
-			ClusterID:         clusterID,
-			ImageCreateParams: &models.ImageCreateParams{},
-		})
-		Expect(err).NotTo(HaveOccurred())
-	})
-
-	It("[V2UpdateCluster] generate_image_after_proxy_was_set", func() {
-		// Generate ISO of registered cluster without proxy configured
-		_, err := userBMClient.Installer.GenerateClusterISO(ctx, &installer.GenerateClusterISOParams{
-			ClusterID:         clusterID,
-			ImageCreateParams: &models.ImageCreateParams{},
-		})
-		Expect(err).NotTo(HaveOccurred())
-
-		// Update cluster with proxy settings
-		httpProxy := "http://proxyserver:3128"
-		noProxy := "test.com"
-		_, err = userBMClient.Installer.V2UpdateCluster(ctx, &installer.V2UpdateClusterParams{
-			ClusterUpdateParams: &models.V2ClusterUpdateParams{
-				HTTPProxy: &httpProxy,
-				NoProxy:   &noProxy,
+		_, err := userBMClient.Installer.UpdateInfraEnv(ctx, &installer.UpdateInfraEnvParams{
+			InfraEnvID: *infraEnvID,
+			InfraEnvUpdateParams: &models.InfraEnvUpdateParams{
+				Proxy: &models.Proxy{
+					HTTPProxy: &httpProxy,
+					NoProxy:   &noProxy,
+				},
 			},
-			ClusterID: clusterID,
 		})
 		Expect(err).ShouldNot(HaveOccurred())
 
-		// Verify proxy settings changed event emitted
-		verifyEventExistence(clusterID, "Proxy settings changed")
+		//Note: Proxy settings changed event is not emitted in V2 API
 
 		// at least 10s must elapse between requests to generate the same ISO
 		time.Sleep(time.Second * 10)
 
-		// Generate ISO of registered cluster with proxy configured
-		_, err = userBMClient.Installer.GenerateClusterISO(ctx, &installer.GenerateClusterISOParams{
-			ClusterID:         clusterID,
-			ImageCreateParams: &models.ImageCreateParams{},
-		})
-		Expect(err).NotTo(HaveOccurred())
+		// Generate infraEnv and ISO with proxy configured
+		_ = registerInfraEnv(&clusterID).ID
```

Comment:
I wonder if we can test something meaningful here.
Was the event the only thing we were really looking for before outside of successful API call?

Reply:
I didn't make a suggestion, I was just wondering if you could think of something else. If not I suppose this is fine.